File: go/gildedrose/gildedrose.go
package gildedrose

type Item struct {
	Name            string
	SellIn, Quality int
}

func UpdateQuality(items []*Item) {

	for i := 0; i < len(items); i++ {
		builderInstance := getBuilder(items[i].Name)

		director := setDirector(builderInstance)
		resultItem := director.setItem(items[i]).eBuilder()

		items[i].SellIn = resultItem.SellIn
		items[i].Quality = resultItem.Quality
	}
}

File: Java/src/main/java/com/gildedrose/GildedRose.java
package com.gildedrose;

class GildedRose {
    Item[] items;

    public GildedRose(Item[] items) {
        this.items = items;
    }

    public void updateQuality() {
        for (Item item : items) {
            ItemEnum itemEnum = ItemEnum.getItemEnum(item.name);
            itemEnum.update(item);
        }
    }
}
